noether_gui/include/noether_gui/widgets.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <QWidget>
+#include <yaml-cpp/yaml.h>
```

Comment:
The intent was to forward-declare as many things as possible to make build times faster, so I don't want to include this file in a header. The issue is that some of the .cpp files don't include `<yaml-cpp/yaml.h>`. I opened #194  to fix that. Presumably this error manifests itself in your build of SNP; we'll have to make a few small changes there as well to address the same issues. I'll open that PR shortly

Reply:
It feels weird to me that a header file would have a reference to YAML::Node but not include the yaml-cpp header. I guess it's fine since it's virtual and the class that inherits from it will include yaml-cpp? Is that what forward-declare means? It feels like it makes implementation confusing

Reply:
Forward declaration is just stating that a class exists without including it's header, as in the lines below this change:

https://github.com/ros-industrial/noether/blob/3d4ceeb84aa4af174009df9bf377e7adc126aad4/noether_gui/include/noether_gui/widgets.h#L5-L16

The header files for those classes are then only included in the source file. This is a fairly common paradigm in large libraries, like Qt, VTK, Boost, etc., and can speed up compile times significantly, especially when many `#include`-ed classes are defined in-line in header files (as in `yaml-cpp`)